### PR TITLE
Add  presentation submission and editing

### PR DESCRIPTION
**Pull request by jberger**
This is still the simple submission flow and no administration stuff like assigning acceptance or scheduling, but that can all come later.

### CODE REVIEW

File: lib/MCT.pm [hunk 1]
```diff
@@ -47,9 +47,14 @@ sub _routes {
...
-  $conf->any('/presentations/:url_name')->to('presentation#')->name('presentation')
-    ->tap(get => {action => 'show'})
-    ->tap(put => {action => 'save'});
+
+  my $presentations = $conf->any('/presentations')->to('presentation#')->name('presentations');
+  $presentations->get('/')->to('#edit');
+  $presentations->post('/')->to('#store');
+  my $presentation = $presentations->any('/:url_name')->name('presentation');
+  $presentation->get('/')->to('#show');
+  $presentation->get('/edit')->to('#edit')->name('presentation.edit');
```
Comment (jhthorsen) [inline lib/MCT.pm:56]: I must say tap() is cool, but it's nice to have it spelled out :smile:

File: lib/MCT/Controller/Presentation.pm [hunk 2]
```diff
@@ -11,17 +11,74 @@ sub show {
...
+  my $p = $c->model->presentation(
+    conference => $c->stash('conference')->identifier,
+    url_name   => $url_name,
+  );
+  $c->delay(
+    sub { $p->load(shift->begin) },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      $c->render_not_authorized unless $p->user_can_update($c->session('username'));
```
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm]: Is this OK without a `return` in front?

File: lib/MCT/Controller/Presentation.pm [hunk 3]
```diff
@@ -11,17 +11,74 @@ sub show {
...
+    },
+  );
+}
+
+sub store {
+  my $c = shift;
+
+  my $id = $c->param('id');
+  my $p = $c->model->presentation(
+    conference => $c->stash('conference')->identifier,
```
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm:46]: Generic question: I use `stash 'cid'`. Do you consider that unsafe or the same?

File: lib/MCT/Controller/Presentation.pm [hunk 4]
```diff
@@ -11,17 +11,74 @@ sub show {
...
+    $id ? (id => $id) : (),
+  );
+
+  my $title  = $c->param('title');
+  my %set = (
+    abstract => $c->param('abstract'),
+    author   => $c->session('username'),
+    subtitle => $c->param('subtitle'),
+    title    => $title,
+    url_name => $c->param('url_name') || $c->_url_name($title),
```
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm:56]: Maybe we should drop "url_name" from param for now, and then just generate the title?

File: lib/MCT/Controller/Presentation.pm [hunk 5]
```diff
@@ -11,17 +11,74 @@ sub show {
...
+  );
+
+  $c->delay(
+    sub { $p->load(shift->begin) },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      $c->render_not_authorized unless $p->user_can_update($c->session('username'));
```
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm]: :+1:

File: lib/MCT/Controller/Presentation.pm [hunk 6]
```diff
@@ -11,17 +11,74 @@ sub show {
...
+    },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      $c->redirect_to(presentation => url_name => $p->url_name);
+    },
+  );
+}
+
+sub render_not_authorized { shift->render(text => 'Not authorized', status => 401) }
```
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm:75]: Sometime in the future we should probably factor this out to `$c->reply->not_authorized`.

File: lib/MCT/Controller/Presentation.pm (hunk 2 above)
Comment (jberger) [inline lib/MCT/Controller/Presentation.pm]: oh, probably not. I probably should test some things without edit perms

File: lib/MCT/Controller/Presentation.pm (hunk 3 above)
Comment (jberger) [inline lib/MCT/Controller/Presentation.pm:46]: we are loading the conference object correct? either is probably fine then.

Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm:46]: OK. Thanks.

File: lib/MCT/Controller/Presentation.pm (hunk 4 above)
Comment (jberger) [inline lib/MCT/Controller/Presentation.pm:56]: what if people want a specific one in the future? I'm willing to agree, but it's not hard to leave it as is.

File: lib/MCT/Controller/Presentation.pm (hunk 5 above)
Comment (jberger) [inline lib/MCT/Controller/Presentation.pm]: this needs the `return` too I guess

File: lib/MCT/Controller/Presentation.pm (hunk 6 above)
Comment (jberger) [inline lib/MCT/Controller/Presentation.pm:75]: I'll add that to the mental todo list

File: lib/MCT/Controller/Presentation.pm (hunk 4 above)
Comment (jhthorsen) [inline lib/MCT/Controller/Presentation.pm:56]: I agree. Leave it as is.

**Comment by jhthorsen** [comment]
Awesome! :+1:

**Pull request merged**